### PR TITLE
Cleanup builder/Dockerfile

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

Cleanup `builder/Dockerfile` and prepare it for containerdisk annotations test. (#417)

- Use quay.io/fedora/fedora:latest as base
- Remove unused dependencies / tools
- Cleanup syntax
- Download latest oc client binary
- Add latest yq binary (needed for containerdisk annotations test)
- Add comments

**Release note**:

```release-note
None
```

### CODE REVIEW

File: builder/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,20 @@
-#This docker file is used in openshift CI
-FROM quay.io/kubevirt/common-templates:fedora33CI
+# This Dockerfile is used in openshift CI
+FROM quay.io/fedora/fedora:latest
```
Comment (akrejcir) [inline builder/Dockerfile:2]: Can you use a fixed version instead of `latest`? Just to be sure that it is not unexpectedly updated.

**Review by akrejcir — COMMENTED** [review]
I'm just curious, why did you separate this PR from https://github.com/kubevirt/common-templates/pull/417 ?

**Comment by 0xFelix** [comment]
@akrejcir: @ksimon1 told me to do so because the CI would not pick up the new `builder/Dockerfile` otherwise. The test introduced #417 depends on it and currently fails.

File: builder/Dockerfile (hunk 1 above)
Comment (0xFelix) [reply builder/Dockerfile:2]: From what I've seen `latest` points to the latest Fedora `stable` release. Ideally, we would always use that anyway? Currently it is an additional maintenance task to update the image whenever a new Fedora `stable` is released.

Comment (akrejcir) [reply builder/Dockerfile:2]: Ok

**Comment by akrejcir** [comment]
> @akrejcir: @ksimon1 told me to do so because the CI would not pick up the new `builder/Dockerfile` otherwise. The test introduced in #417 depends on it and currently fails.

All right, I didn't know that.

File: builder/Dockerfile [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-# download oc client binary
-RUN curl https://mirror2.openshift.com/pub/openshift-v4/clients/oc/4.6/linux/oc.tar.gz | tar -C . -xzf -
-RUN chmod +x oc
-RUN mv oc /usr/bin
+# Create ansible tmp folder and set permissions
+RUN mkdir -p /.ansible/tmp && \
+    chmod -R 777 /.ansible
+
+# Download oc client binary
+RUN curl https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/oc/latest/linux/oc.tar.gz | tar -C /usr/local/bin -xzf - oc && \
```
Comment (ksimon1) [inline builder/Dockerfile:15]: We might go into troubles with latest oc in the future. Please use the same oc client version as we are using for openshift in e2e tests

**Pull request merged**